src/api/entities/MetadataEntry/index.ts

```diff
@@ -0,0 +1,178 @@
+  }
+
+  /**
+   * @hidden
+   */
+  public constructor(identifiers: UniqueIdentifiers, context: Context) {
+    super(identifiers, context);
+
+    const { ticker, type, id } = identifiers;
+
+    this.asset = new Asset({ ticker }, context);
+    this.type = type;
+    this.id = id;
+
+    this.set = createProcedureMethod(
+      { getProcedureAndArgs: args => [setMetadata, { ...args, metadataEntry: this }] },
+      context
+    );
+  }
+
+  /**
+   * Assign new value for the MetadataEntry along with its details or optionally only set the details (expiry + lock status) of any Metadata value
+   *
+   * @note - Value or the details can only be set if the MetadataEntry is not locked
+   */
+  public set: ProcedureMethod<SetMetadataParams, MetadataEntry>;
+
+  /**
+   * Retrieve name and specs for this MetadataEntry
+   */
+  public async details(): Promise<MetadataDetails> {
+    const {
+      context: {
+        polymeshApi: {
+          query: {
+            asset: {
+              assetMetadataLocalKeyToName,
+              assetMetadataLocalSpecs,
+              assetMetadataGlobalKeyToName,
+              assetMetadataGlobalSpecs,
+            },
+          },
+        },
+      },
+      id,
+      asset: { ticker },
+      type,
+      context,
+    } = this;
+
+    const rawId = bigNumberToU64(id, context);
+    const rawTicker = stringToTicker(ticker, context);
+
+    let rawName, rawSpecs;
+    if (type === MetadataType.Local) {
+      [rawName, rawSpecs] = await Promise.all([
+        assetMetadataLocalKeyToName(rawTicker, rawId),
+        assetMetadataLocalSpecs(rawTicker, rawId),
+      ]);
+    } else {
+      [rawName, rawSpecs] = await Promise.all([
+        assetMetadataGlobalKeyToName(rawId),
+        assetMetadataGlobalSpecs(rawId),
+      ]);
+    }
+
+    return {
+      name: bytesToString(rawName.unwrap()),
+      specs: meshMetadataSpecToMetadataSpec(rawSpecs),
+    };
+  }
+
+  /**
+   * Retrieve the value and details (expiry + lock status) for this MetadataEntry
+   *
+   * @note - This returns `null` if no value is yet specified for this MetadataEntry
+   */
+  public async value(): Promise<MetadataValue | null> {
+    const {
+      context: {
+        polymeshApi: {
+          query: {
+            asset: { assetMetadataValues, assetMetadataValueDetails },
+          },
+        },
+      },
+      id,
+      type,
+      asset: { ticker },
+      context,
+    } = this;
+
+    const rawMetadataKey = metadataToMeshMetadataKey(type, id, context);
+    const rawTicker = stringToTicker(ticker, context);
+
+    const [rawValue, rawValueDetails] = await Promise.all([
+      assetMetadataValues(rawTicker, rawMetadataKey),
+      assetMetadataValueDetails(rawTicker, rawMetadataKey),
+    ]);
+
+    return meshMetadataValueToMetadataValue(rawValue, rawValueDetails);
+  }
+
+  /**
+   * Determine whether this MetadataEntry exists on chain
+   */
+  public async exists(): Promise<boolean> {
+    return true;
```

Comment:
Is it able to delete a MetadataEntry once configured?

Reply:
Not right now